lib/archive/unpack_unix.go

```diff
@@ -202,6 +202,56 @@ func streamCopy(op trace.Operation, stdin io.WriteCloser, tarStream io.Reader) e
 	}
 }
 
+func DockerUnpack(op trace.Operation, root string, tarStream io.Reader) (int64, error) {
+	fi, err := os.Stat(root)
+	if err != nil {
+		// the target unpack path does not exist. We should not get here.
+		op.Errorf("tar unpack target does not exist: %s", root)
```

Comment:
You may get this error not only because file doesn't not exist, there permission reasons and other potential error. You may want to check for NotExists error specifically. before you write a log message.

Reply:
yeah, if we get an error here, we should just report the error regardless of what kind it is -- I'll update this. I'll also update the other parts of the file this same error reporting routine is used, because I totally copied this part from old code, ha.